build/maven_test.go

```diff
@@ -72,3 +72,52 @@ func getExpectedMavenBuildInfo(t *testing.T, filePath string) entities.BuildInfo
 	assert.NoError(t, json.Unmarshal(data, &buildinfo))
 	return buildinfo
 }
+
+func TestExtractMavenPath(t *testing.T) {
+	service := NewBuildInfoService()
+	mavenBuild, err := service.GetOrCreateBuild("build-info-maven-test", "1")
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, mavenBuild.Clean())
+	}()
+	testdataDir, err := filepath.Abs(filepath.Join("testdata"))
+	assert.NoError(t, err)
+	// Create maven project
+	projectPath := filepath.Join(testdataDir, "maven", "project")
+	tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+	defer cleanup()
+	// Add maven project as module in build-info.
+	mavenModule, err := mavenBuild.AddMavenModule(tmpProjectPath)
+	assert.NoError(t, err)
+
+	s1 := "Maven home: /test/is/good"
+	s2 := "Home: /test/is/not/good"
+	s3 := "Mvn Home:= /test/is/not/good"
+	var output []string
+	output = append(output, s1, s2, s3)
+	mavenHome, err := mavenModule.ExtractMavenPath(output)
+	assert.NoError(t, err)
+	if utils.IsWindows() {
+		assert.Equal(t, "/test/is/good/r", mavenHome)
+	} else {
+		assert.Equal(t, "/test/is/good", mavenHome)
+	}
+	output = nil
+	output = append(output, s2, s1, s3)
+	mavenHome, err = mavenModule.ExtractMavenPath(output)
+	assert.NoError(t, err)
+	if utils.IsWindows() {
+		assert.Equal(t, "/test/is/good/r", mavenHome)
+	} else {
+		assert.Equal(t, "/test/is/good", mavenHome)
+	}
+	output = nil
+	output = append(output, s2, s3)
+	mavenHome, err = mavenModule.ExtractMavenPath(output)
+	if utils.IsWindows() {
+		assert.Equal(t, "/r", mavenHome)
+	} else {
+		assert.Equal(t, "", mavenHome)
+	}
+	assert.NoError(t, err)
```

Comment:
This requires table testing to remove code duplication. See how we do in TestGenerateBuildInfoForPython

I also find this guide useful:
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Done

Reply:
Done